src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return this[0].toString();
+    }
+
+    let thisButStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        thisButStr += '';
+      } else {
+        thisButStr += this[i];
+      }
+
+      if (i < this.length - 1) {
```

Comment:
in that case better use `!==` operator because you`re avoiding only one specific value.

Reply:
> in that case better use `!==` operator because you`re avoiding only one specific value.

why? if i use i < this.length - 1 - i check every time is it and if i use i !== this.length - 1 i check every time is it

Reply:
`< operator ` would work too, but in my opinion, it's better to check only `this.length - 1` value
P.S. not `this.length` or `this.length + 3` and etc.

Reply:
i check only this.length - 1 value